src/Components/Server/src/Circuits/RemoteJSRuntime.cs

```diff
@@ -48,6 +49,13 @@ internal partial class RemoteJSRuntime : JSRuntime
         DefaultAsyncTimeout = _options.JSInteropDefaultCallTimeout;
         ElementReferenceContext = new WebElementReferenceContext(this);
         JsonSerializerOptions.Converters.Add(new ElementReferenceJsonConverter(ElementReferenceContext));
+
+        JsonSerializerOptions.TypeInfoResolverChain.Add(JsonConverterFactoryTypeInfoResolver.Instance);
+
+        if (jsonOptions.Value.SerializerOptions is { TypeInfoResolver: { } typeInfoResolver })
+        {
+            JsonSerializerOptions.TypeInfoResolverChain.Add(typeInfoResolver);
+        }
```

Comment:
I think we need to define a context for our internal types and use that here, otherwise people could hijack `ElementReference` by providing their own in the typeInfoResolver chain (I believe). At least we need to make sure that this doesn't allow changing the serialization semantics of the core bits of the protocol.